src/main/java/uk/gov/hmcts/darts/event/service/impl/EventProcessorImpl.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.darts.event.service.impl;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import uk.gov.hmcts.darts.common.helper.CurrentTimeHelper;
+import uk.gov.hmcts.darts.common.repository.EventRepository;
+import uk.gov.hmcts.darts.event.service.EventProcessor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+@RequiredArgsConstructor
+public class EventProcessorImpl implements EventProcessor {
+    private final Integer batchSize;
+    private final EventRepository eventRepository;
+    private final CurrentTimeHelper currentOffsetDateTime;
+
+    @Override
+    public List<Integer> processCurrentEvent() {
+        List<Integer> currentEventLst = new ArrayList<>();
+        List<Integer> eventEntityReturned = eventRepository.getCurrentEventIdsToBeProcessed(batchSize);
+        eventEntityReturned.stream().forEach(event -> {
```

Comment:
for a batch size of X this would result in 2X distinct sql statements to the DB which can be a bit slow. I was wondering if it's possible to remove this loop and instead send the DB 1 or 2 sqls only that update in one go all the necessary rows

Reply:
i'll look but your only going to get batch size at most which shouldnt be that large.